### PR TITLE
Add taxonomy

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
Local compilation: https://travis-ci.org/fxbox/foxbox/jobs/124767230#L824
Tests run: https://travis-ci.org/fxbox/foxbox/jobs/124767230#L1371

r? @Yoric

**Comment by mcav** [comment]
What will happen to Taxonomy's WIP branches (e.g. https://github.com/fxbox/taxonomy/issues/104 ) when we merge this?

**Comment by Yoric** [comment]
We'll need to migrate them manually. @JohanLorenzo  has promised to help :)

**Comment by JohanLorenzo** [comment]
Yes, we'll have to manually create the new branches/PRs. A couple of ways I see:
- Generate a diff and apply it to the new repo
- `git filter-branch` to move the files to the correct path, pull the commit in the new repo (but not in a branch yet) and cherry-pick the commits in the right branch.

File: components/taxonomy/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+target/
```
Comment (Yoric) [inline components/taxonomy/.gitignore]: I haven't paid attention for other repos, but is this file necessary?

**Comment by Yoric** [comment]
r=me

File: components/taxonomy/.gitignore (hunk 1 above)
Comment (JohanLorenzo) [inline components/taxonomy/.gitignore]: Good catch. That particular file is not necessary as `.gitignore` at the root of the repo already takes care of `target/` and `Cargo.lock`. I added an extra commit to clean it up
